src/services/wallet-service/wallet-service.spec.ts

```diff
@@ -0,0 +1,201 @@
+                    {"id":"9b5329ff-c683-42a5-9165-4093e4076166",
+                        "type":"local",
+                        "name":"Labilo",
+                        "publicKey":"ELsKCchf9rcGsufjRR62PG5Fn5dFinfgeN",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"},
+                    {"id":"a2e16167-fedb-47d2-8856-2b3f97389c35",
+                        "type":"local",
+                        "name":"Zalista",
+                        "publicKey":"EZ7tP3CBdBKrB9MaBgZNHyDcTg5TFRRpaY",
+                        "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"}]);
+                done();
+            });
+        });
+    });
+
+    it("should return cloned wallets when there are already existing wallets", () => {
+        spyOn(storage, "get").and.callThrough();
+        (<any>walletService).wallets = [
+            {"id":"012d294e-cb11-439b-937a-12d47a52c305",
+                "type":"local",
+                "name":"Biosta",
+                "publicKey":
+                "ETm9QUJLVdJkTqRojTNqswmeAQGaofojJJ",
+                "encryptedPrivateKey":"E9873D79C6D87DC0FB6A5778633389F4453213303DA61F20BD67FC233AA33262"}];
+        walletService.getAll();
+        expect(storage.get).not.toHaveBeenCalled();
+    });
+
+    it("should push a new wallet when it does not yet exist and overwrite when it already exists", (done) => {
+        // So that getWalletIndex doesn't crash on undefined length
+        (<any>walletService).wallets = [];
+        // Let to reset later
+        let pushSpy = spyOn((<any>walletService).wallets, "push").and.callThrough();
+        spyOn(storage, "set").and.callThrough();
+        let wallet = <IWallet>{id : "4b6cff11-5888-43bb-bde1-911e12b659e6",
+            keyStore: { 
+                cipher: "AES-CTR",
+                cipherParams: {
+                    iv: "a/ÿûÅ)rêYgÅ.¾DÖwW;6×aqr"
+                },
+                cipherText : "JIH", 
+                controlHash : "e845922979b1fad26a716ac155a4cbb822c6538561d7e575206190e87200d4c7",
+                keyParams: {
+                    iterations: 128,
+                    keySize: 32,
+                    alt: "G'G&ÃµÈ¶qvÍO£M3ý`~põqög` ↵¯4\¤BùøÃ{!êjô\Ý ½ÎêN«Î¥^²Ôô`LEK_0{×ôºæç¯FÉdÒ`6ÉSîK¬D¡Ün Û¡c¸Éz↵ë*P$}Lò?%±à$Ù¿BëÒ<@dT3'ê Xï¡ cÿÑÎÉ~5¶÷ûS@ù/¡ö+°¿BCÍêüf↵ÑÃ&öêX](<ä=AósµcU£éÒÀæÿ:¡íÓN+¹Py`ÿÈË5H1ÑRï¼" 
+                },
+            },
+            lastUpdateTime: null,
+            name: "Bosha",
+            publicKey : "S5NEKHPKXS7F75IVKGVS4A56U4FF6VM5U4YF64",
+            type : "local"}
+        // Try store new wallet
+        walletService.store(wallet).then(data => {
+            // Pushed on the array
+            expect((<any>walletService).wallets.push).toHaveBeenCalledWith(wallet);
```

Comment:
As discussed it might be better to test the `wallets` array to see if the wallet was removed instead of testing if a function which will remove the wallet from the array was called.

Same goes for the next test which tests the splice method.